src/field_theory/krull_topology.lean

```diff
@@ -283,3 +285,219 @@ begin
+  exact intermediate_field.to_subalgebra_eq_iff.mp (subalgebra.to_submodule_injective
+    (finite_dimensional.eq_of_le_of_finrank_eq h_map_le h_finrank_eq.symm)),
+end
+
+lemma ultrafilter.glued_generators_of_pushforwards_alg_hom_surjective (h_int :
+  algebra.is_integral K L) (f : ultrafilter (L →ₐ[K] L)) :
+  function.surjective (ultrafilter.glued_generators_of_pushforwards_alg_hom f h_int) :=
+begin
+  intro y,
+  let p := minpoly K y,
+  let S := p.root_set L,
+  let σ := ultrafilter.glued_generators_of_pushforwards_alg_hom f h_int,
+  have hp : p ≠ 0 := (minpoly.monic (h_int y)).ne_zero,
+  have hσSS : σ '' S ⊆ S,
+  { rintros x ⟨a, ha, rfl⟩,
+    rw polynomial.mem_root_set hp at ha,
+    rw [polynomial.mem_root_set hp, polynomial.aeval_alg_hom_apply, ha, σ.map_zero] },
+  have h3 : σ '' S = S :=
+  eq_of_subset_of_card_le hσSS (ge_of_eq (card_image_of_injective S σ.to_ring_hom.injective)),
+  have hy : y ∈ σ '' S,
+  { rw [h3, polynomial.mem_root_set hp, minpoly.aeval] },
+  rcases hy with ⟨a, ha, hay⟩,
+  exact ⟨a, hay⟩,
+end
+
+lemma ultrafilter.glued_generators_of_pushforwards_alg_hom_bijection (h_int : algebra.is_integral
+  K L) (f : ultrafilter (L →ₐ[K] L)) : function.bijective
+  (ultrafilter.glued_generators_of_pushforwards_alg_hom f h_int) :=
+⟨ultrafilter.glued_generators_of_pushforwards_alg_hom_injective h_int f,
+  ultrafilter.glued_generators_of_pushforwards_alg_hom_surjective h_int f⟩
+
+/-- Since `ultrafilter.glued_generators_of_pushforwards_alg_hom h_int f` is a bijective `K`-algebra
+  homomorphism `L →ₐ[K] L`,it is a `K`-algebra equivalence. Here we define the corresponding term
+  of `L ≃ₐ[K] L`. -/
+noncomputable def ultrafilter.glued_generators_of_pushforwards_alg_equiv (h_int :
+  algebra.is_integral K L) (f : ultrafilter (L →ₐ[K] L)) : (L ≃ₐ[K] L) :=
+alg_equiv.of_bijective (ultrafilter.glued_generators_of_pushforwards_alg_hom f h_int)
+  (ultrafilter.glued_generators_of_pushforwards_alg_hom_bijection h_int f)
+
+/-- Let `L/K` be a normal algebraic field extension, let `f` be an ultrafilter on
+  `L ≃ₐ[K] L`, and let `E/K` be a finite subextension. Then
+  `ultrafilter.glued_generators_of_pushforwards_alg_equiv h_int f`
+  is a term of `L ≃ₐ[K] L`, and `alg_hom_of_finite_dimensional_of_ultrafilter h_findim f` is a term
+  `E →ₐ[K] L`. This Lemma tells us that the underlying maps of these two terms agree on `E`.
+    -/
+lemma equiv_restricts_to_alg_hom_of_finite_dimensional (h_int : algebra.is_integral K L)
+  (f : ultrafilter (L →ₐ[K] L)) {E : intermediate_field K L} (h_findim : finite_dimensional K E) :
+  ((ultrafilter.glued_generators_of_pushforwards_alg_equiv h_int f).to_alg_hom.comp E.val) =
+  f.generator_of_pushforward h_findim :=
+alg_hom.ext (ultrafilter.glued_generators_of_pushforwards_function_spec' h_int f h_findim)
+
+lemma ultrafilter_converges_to_glued_equiv (h_int : algebra.is_integral K L)
+  (f : ultrafilter (L ≃ₐ[K] L)) : (f : filter (L ≃ₐ[K] L)) ≤ nhds
```

Comment:
Don't we have syntax for `nhds`?

Reply:
I'm not sure what you mean by this. Could you elaborate?

Reply:
I think there's an `open_locale` that allows you to write nhds with a fancy `N`. It's all over the topology part of the library.

Reply:
We have notation `𝓝` in locale `topological_space`.

Reply:
@Sebastian-Monnet can you resolve this if you've dealt with it?